---
title: "21_obtaining_TPM_for_DEGs.Rmd"
author: "Aidan Coyle"
date: "Last compiled on `r format(Sys.time(), '%Y-%m-%d')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Aidan Coyle, afcoyle@uw.edu

Roberts Lab, UW-SAFS

Originally written 2021-02-16


This script takes a file of genes that are differentially expressed between two groups (obtained from DESeq2) and cross-references it with the transcriptome used to create your kallisto index. It then extracts the TPM (transcripts per million) counts from the kallisto libraries created earlier in the pipeline.

This produces a table containing the following:

- The transcript and gene IDs for all inputted DEGs
- All TPM counts from individual kallisto libraries for those DEGs

First, loading the required libraries
```{r libraries, message = FALSE, warning=FALSE}
# Add all required libraries here
list.of.packages <- "tidyverse"
# Get names of all required packages that aren't installed
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[, "Package"])]
# Install all new packages
if(length(new.packages)) install.packages(new.packages)


# Load all required libraries
lapply(list.of.packages, FUN = function(X) {
  do.call("require", list(X))
})
```

Then, we edit our variables. If running this script on other samples, you only need to edit the following section.

The following inputs are required

DEG_list: the filepath to your table of differentially-expressed genes
blast_data: the filepath to the transcriptome. Should be the same transcriptome used to create the kallisto index
outpath: the file you want to write the results to
kallisto_path: the path to the folder containing all your kallisto libraries
libraries: specify the libraries you want to add to your final matrix of TPM. Using the notation in Grace's hematodinium/bairdi    libraries, the following notation applies (utilizes wildcards):

- Individual libraries only: ???
- Pooled libraries only: ??????
- All libraries: *

```{r}

# Import DEG list outputted from DESeq2
DEG_list <- read.table("../graphs/DESeq2_output/cbai_transcriptomev2.0/amb2_vs_elev2_indiv/DEGlist_wcols.txt",
                       header = TRUE,
                       sep = "\t")

blast_data <- read.delim("../data/cbai_diamond_blastx_table_transcriptomev2.0.txt", 
                         header = FALSE,
                         sep = "\t")

# Specify file you want to write results to
outpath <- "../output/TPM_counts/cbai_transcriptomev2.0/amb2_vs_elev2_DEG_TPMs.txt"

kallisto_path <- "../output/kallisto_libraries/cbai_transcriptomev2.0/"

libraries <- "???"

```

We then cross-reference the DEG table with our BLAST data to create a two-column table of transcript IDs and gene IDs

```{r}
# Transcript IDs are rownames - move them into first column
DEG_list <- rownames_to_column(DEG_list,
                               "Transcript_ID")

# Remove all columns that aren't transcript ID
DEG_list <- DEG_list[,1, drop = FALSE]

# Columns have no names - add names for first two columns
colnames(blast_data)[1:2] <- c("Transcript_ID", "Gene_ID")

# Turn the first two columns of BLAST data into a Transcript ID/Gene ID key
blast_data <- blast_data %>%
  select(Transcript_ID, Gene_ID)

# Add Gene ID column to transcript data, using Transcript ID column to match
DEG_list <- left_join(DEG_list, blast_data, by = "Transcript_ID")

# Select only DEGs with transcript IDs that match to genes
DEG_list <- DEG_list[!is.na(DEG_list$Gene_ID),]

head(DEG_list)


```

We then extract the filenames for all kallisto files we want to match, and add a TPM column to our table for each kallisto file that we want to match

```{r}
kallisto_filepath <- paste0(kallisto_path, "id", libraries, "/abundance.tsv")
# List all kallisto indices
kallisto_files <- Sys.glob(kallisto_filepath)


for (i in 1:length(kallisto_files)) {
  # Extract the ID number from the kallisto file
  idnum <- str_extract(kallisto_files[i], "id[0-9]+")
  # Read in the kallisto file
  kallisto_output <- read.delim(file = kallisto_files[i], 
                                header = TRUE,
                                sep = "\t")
  # Select only transcript ID and TPM (transcripts per million) columns
  kallisto_output <- kallisto_output %>%
    select(target_id, tpm)
  
  # Rename kallisto column names
  colnames(kallisto_output)[1:2] <- c("Transcript_ID", 
                                      paste0(idnum, "_TPM"))
  # Add TPM value to table of DEGs
  DEG_list <- left_join(DEG_list, kallisto_output, by = "Transcript_ID")
}

head(DEG_list)

# Write results to table
write.table(DEG_list,
            file = outpath,
            quote = FALSE,
            row.names = FALSE)
```
